ui/src/components/MotorInput/TwoAxisTranslationControl.jsx

```diff
@@ -1,137 +1,153 @@
-          className="arrow arrow-left"
-          disabled={
-            this.props.motors.sample_horizontal.state !== HW_STATE.READY ||
-            this.props.motorsDisabled
-          }
-          onClick={() =>
-            this.stepChange('sample_horizontal', sample_horizontalStep, -1)
-          }
-        >
-          <i className="fas fa-angle-left" />
-        </Button>
-        <OverlayTrigger
-          trigger="click"
-          rootClose
-          placement="right"
-          overlay={this.renderMotorSettings()}
-        >
-          <Button
-            size="sm"
-            variant="outline-secondary"
-            className="arrow arrow-settings"
-          >
-            <i className="fas fa-cog" />
-          </Button>
-        </OverlayTrigger>
-        <Button
-          size="sm"
-          variant="outline-secondary"
-          className="arrow arrow-right"
-          disabled={
-            this.props.motors.sample_horizontal.state !== HW_STATE.READY ||
-            this.props.motorsDisabled
-          }
-          onClick={() =>
-            this.stepChange('sample_horizontal', sample_horizontalStep, 1)
-          }
-        >
-          <i className="fas fa-angle-right" />
-        </Button>
+  const motorsDisabled = useSelector(
+    (state) =>
+      state.beamline.motorInputDisable ||
+      state.queue.queueStatus === QUEUE_RUNNING,
+  );
+
+  return (
+    <div className="arrow-control mb-3">
+      <p className="motor-name mb-2">Sample alignment:</p>
+
+      <Button
+        size="sm"
+        variant="outline-secondary"
+        onClick={() =>
+          dispatch(
+            setAttribute(
+              verticalMotorProps.attribute,
+              verticalMotor.value + sample_verticalStep,
+            ),
+          )
+        }
+        disabled={motorsDisabled || verticalMotor.state !== HW_STATE.READY}
+        className="arrow arrow-up"
+      >
+        <i className="fas fa-angle-up" />
+      </Button>
+      <Button
+        size="sm"
+        variant="outline-secondary"
+        className="arrow arrow-left"
+        disabled={motorsDisabled || horizontalMotor.state !== HW_STATE.READY}
+        onClick={() =>
+          dispatch(
+            setAttribute(
+              horizontalMotorProps.attribute,
+              horizontalMotor.value - sample_horizontalStep,
+            ),
+          )
+        }
+      >
+        <i className="fas fa-angle-left" />
+      </Button>
+      <OverlayTrigger
+        trigger="click"
+        rootClose
+        placement="right"
+        overlay={
+          <Popover>
+            <Popover.Header as="h3">Sample alignment motors</Popover.Header>
+            <Popover.Body>
+              <MotorInput
+                save={(name, val) => dispatch(setAttribute(name, val))}
+                value={verticalMotor.value}
+                saveStep={(name, val) => dispatch(setStepSize(name, val))}
+                step={verticalMotorProps.step}
```

Comment:
I think I found why changing the steps of the sample vertical/horizontal motors doesn't work: the value in the step input is read from `state.uiproperties`, but the up/down/left/right buttons read the step from `state.sampleview.motorSteps`.

The `SET_STEP_SIZE` action does save the step in these two places, but when it's read from `state.sampleview.motorSteps`, we use hardcoded keys instead of the motor names:

```js
// Before refactoring
const { sample_verticalStep, sample_horizontalStep } = this.props.steps;

// After refactoring
const { sample_verticalStep, sample_horizontalStep } = useSelector(
  (state) => state.sampleview.motorSteps,
);
```

The `sampleview` reducer stores the updated step in `motorSteps` against the name of the motor (e.g. `diffractometer.phiy`), not against `sample_verticalStep` and `sample_horizontalStep`, so the cross never gets the updated values.

Should I strive to remove this `sampleview.motorSteps` state in my next PR? Or on the contrary, should the step always be read from `sampleview.motorSteps` instead of `uiproperties`?

- One could argue that `uiproperties` is a server state that shouldn't be modified locally, in which case accessing it via `motorSteps` makes sense.
- On the other hand, perhaps the step state shouldn't be a local state but a server state, in which case accessing it via `uiproperites` makes sense (as long as it's updated by refetching the latest `uiproperties` state from the server).

Reply:
Thanks very nice as always :), I think removing `sampleview.motorSteps` seems like a good idea as well as regarding`uiproperites` as a server state. Unless I'm overlooking something.

Reply:
Ideally, `uiproperties` should not be modified locally – it should always match the state of `uiproperties` on the server, so if we want to keep it as the source of truth for the step size, there should be a route to update the step size in `uiproperties`.

Since this decision has other implications (observer mode, persistence between users, etc.), and since `motorSteps` seems to be the legacy implementation, I reckon I'll remove `motorSteps` for now and keep modifying the step size locally in `uiproperties` for now. I'll add a comment in the code so we remember to revisit this later.